coriolis/tests/osmorphing/test_windows.py

```diff
@@ -0,0 +1,893 @@
+                ignore_stdout=True),
+            mock.call(
+                "mkdir '%s' -Force" %
+                mock_get_cbslinit_scripts_dir.return_value,
+                ignore_stdout=True),
+        ])
+
+        conf_file_path = (
+            "%s\\conf\\cloudbase-init.conf" %
+            mock.sentinel.cloudbaseinit_base_dir)
+        conf_content = (
+            "[DEFAULT]\r\n"
+            "username = Admin\r\n"
+            "groups = Administrators\r\n"
+            "verbose = true\r\n"
+            "bsdtar_path = %(bin_path)s\\bsdtar.exe\r\n"
+            "mtools_path = %(bin_path)s\r\n"
+            "logdir = %(log_path)s\r\n"
+            "local_scripts_path = %(scripts_path)s\r\n"
+            "stop_service_on_exit = false\r\n"
+            "logfile = cloudbase-init.log\r\n"
+            "default_log_levels = \r\n"
+            "comtypes=INFO,suds=INFO,iso8601=WARN,requests=WARN\r\n"
+            "allow_reboot = false\r\n"
+            "plugins = %(plugins)s\r\n"
+            "debug = true\r\n"
+            "san_policy = OnlineAll\r\n"
+            "metadata_services = %(metadata_services)s\r\n"
+            "logging_serial_port_settings = %(com_port)s,9600,N,8\r\n" %
+            {"bin_path": "%s\\Bin" % mock.sentinel.local_base_dir,
+             "log_path": "%s\\Log" % mock.sentinel.local_base_dir,
+             "scripts_path": "%s\\LocalScripts" % mock.sentinel.local_base_dir,
+             "com_port": 'COM1',
+             "metadata_services": ",".join(
+                 windows.CLOUDBASE_INIT_DEFAULT_METADATA_SVCS),
+             "plugins": ",".join(windows.CLOUDBASE_INIT_DEFAULT_PLUGINS)})
+
+        mock_write_winrm_file.assert_called_once_with(
+            self.conn, conf_file_path, conf_content)
+
+        mock_write_local_script.assert_called_once_with(
+            mock.sentinel.cloudbaseinit_base_dir,
+            mock_os_path.join.return_value, priority=99)
+
+    @mock.patch.object(
+        windows.BaseWindowsMorphingTools, '_get_cbslinit_scripts_dir'
+    )
+    @mock.patch.object(windows.utils, 'write_winrm_file')
+    @mock.patch.object(windows.BaseWindowsMorphingTools, '_write_local_script')
+    def test__write_cloudbase_init_conf_with_exception(
+            self, mock_write_local_script, mock_write_winrm_file,
+            mock_get_cbslinit_scripts_dir):
+        plugins = "invalid plugins"
+
+        self.assertRaises(
+            exception.CoriolisException,
+            self.morphing_tools._write_cloudbase_init_conf,
+            mock.sentinel.cloudbaseinit_base_dir, mock.sentinel.local_base_dir,
+            com_port='COM1', plugins=plugins)
+
+        mock_get_cbslinit_scripts_dir.assert_not_called()
+        self.conn.exec_ps_command.assert_not_called()
+        mock_write_winrm_file.assert_not_called()
+        mock_write_local_script.assert_not_called()
+
+    @mock.patch.object(
+        windows.BaseWindowsMorphingTools, '_get_cbslinit_base_dir'
```

Comment:
You don't need to mock this. You can do assertions based on the actual value of this method.